infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/transferprocessor/ParallelTransferProcessor.java

```diff
@@ -32,10 +32,23 @@
     /**
      * A number of in-flight requests per one node.
      */
-    private static final int NUM_REQUESTS_PER_NODE = 5;
+    private final int batchesPerNode;
+
+    /**
+     * A Number of addresses per node.
+     */
+    private static final int DEFAULT_THROUGHPUT_PER_NODE = 500;
 
     public ParallelTransferProcessor(StateTransferBatchProcessor stateTransferBatchProcessor) {
         this.stateTransferBatchProcessor = stateTransferBatchProcessor;
+        this.batchesPerNode = 5;
+    }
+
+    public ParallelTransferProcessor(StateTransferBatchProcessor stateTransferBatchProcessor,
+                                     int batchSize) {
+        this.stateTransferBatchProcessor = stateTransferBatchProcessor;
+        verifyBatchSize(batchSize);
+        this.batchesPerNode = DEFAULT_THROUGHPUT_PER_NODE / batchSize;
```

Comment:
Going by this calculation, there will be 500/20=25 concurrent requests for a single node, which I think maybe too large. I guess you did this to match the existing throughput, but I think we might not be able to make it the same when reducing the batchsize.

Reply:
fixed